File: Skattata.Core/SieVoucher.cs
namespace Skattata.Core;

public class SieVoucher
{
    public SieVoucher()
    {
        Rows = new List<SieVoucherRow>();
    }
    public string Series { get; set; } = "";
    public string Number { get; set; } = "";
    public DateTime Date { get; set; }
    public string Text { get; set; } = "";
    public DateTime RegistrationDate { get; set; }
    public string RegistrationSign { get; set; } = "";

    public List<SieVoucherRow> Rows { get; }
    
    public decimal Balance => Rows.Sum(r => r.Amount);
}

File: packages/sie-core/src/models/SieVoucher.ts
import { SieVoucherRow } from './SieVoucherRow.js';

export class SieVoucher {
  series: string = '';
  number: string = '';
  date: Date = new Date(0);
  text: string = '';
  registrationDate: Date | null = null;
  registrationSign: string = '';
  rows: SieVoucherRow[] = [];

  get balance(): number {
    return this.rows.reduce((sum, r) => sum + r.amount, 0);
  }
}
